src/mpi/coll/iallgather/iallgather_tsp_brucks_algos.h

```diff
@@ -85,7 +86,8 @@ MPIR_TSP_Iallgather_sched_intra_brucks(const void *sendbuf, int sendcount,
     if (MPL_ipow(k, nphases) == size)
         p_of_k = 1;
 
-    recv_id = (int *) MPL_malloc(sizeof(int) * nphases * (k - 1), MPL_MEM_COLL);
+    MPIR_CHKLMEM_MALLOC(recv_id, int *, sizeof(int) * nphases * (k - 1),
+                        mpi_errno, "recv_id buffer", MPL_MEM_COLL);
```

Comment:
Are we doing this because it is too complicated for us to correctly balancing out malloc and free?

Reply:
No, allocating this way gives proper error messages in case of failure.